src/server/lib/user-features.ts

```diff
@@ -42,3 +60,47 @@ export const setUserFeatureCookies = async ({
 		});
 	}
 };
+
+export const createCookie = ({
+	name,
+	res,
+	daysTillExpiry,
+}: {
+	name: string;
+	res: Response;
+	daysTillExpiry: number;
+}) => {
+	const tmpDate = new Date(Date.now() + daysTillExpiry * 24 * 60 * 60 * 1000);
+	res.cookie(name, tmpDate.getTime().toString(), {
+		domain,
+		expires: tmpDate,
+	});
+};
+
+export const persistUserBenefitsCookies = ({
+	userBenefits,
+	res,
+}: {
+	userBenefits: UserBenefitsResponse;
+	res: Response;
+}) => {
+	if (userBenefits?.benefits?.includes('hideSupportMessaging')) {
+		createCookie({
+			name: HIDE_SUPPORT_MESSAGING_COOKIE,
+			res,
+			daysTillExpiry: 7,
+		});
+	}
+	if (userBenefits?.benefits?.includes('allowRejectAll')) {
+		createCookie({ name: ALLOW_REJECT_ALL_COOKIE, res, daysTillExpiry: 7 });
+	}
+	if (userBenefits?.benefits?.includes('adFree')) {
+		createCookie({ name: AD_FREE_USER_COOKIE, res, daysTillExpiry: 2 });
+	}
+
+	createCookie({
+		name: USER_BENEFITS_EXPIRY_COOKIE,
+		res,
+		daysTillExpiry: 7,
+	});
```

Comment:
And a comment for this cookie as well to link to it's correstonding definition in DCR!

I also think that we should only add this cookie only if we have at least 1 benefit we're setting, if we're not setting any benefits, then I don't think we need to add this?

Reply:
This need to be set once the request o user benefits is made. 

In a scenario where a user doesn't have any benefits. We still want to drop the cookie so it doesn't make a request every time they view the site. The user benefits expires after 24 hours. https://github.com/guardian/dotcom-rendering/blob/a0eff55ee7fedf08b785557fceaa0f4f7265e0df/dotcom-rendering/src/client/userFeatures/user-features.ts#L38